specification/databoxedge/resource-manager/Microsoft.DataBoxEdge/preview/2020-09-01-preview/databoxedge.json

```diff
@@ -0,0 +1,8511 @@
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "addon",
+            "in": "body",
+            "description": "The addon properties.",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/Addon"
+            }
+          },
+          {
+            "$ref": "#/parameters/subscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/resourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          }
+        ],
+        "responses": {
+          "202": {
+            "description": "Accepted the request to create or update the addon."
+          },
+          "200": {
+            "description": "Successfully created or updated the addon.",
+            "schema": {
+              "$ref": "#/definitions/Addon"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-examples": {
+          "PutAddons": {
+            "$ref": "./examples/PutAddons.json"
+          }
+        }
+      },
+      "delete": {
+        "tags": [
+          "Addons"
+        ],
+        "description": "Deletes the addon on the device.",
+        "operationId": "Addons_Delete",
+        "parameters": [
+          {
+            "name": "deviceName",
+            "in": "path",
+            "description": "The device name.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "roleName",
+            "in": "path",
+            "description": "The role name.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "addonName",
+            "in": "path",
+            "description": "The addon name.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "$ref": "#/parameters/subscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/resourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          }
+        ],
+        "responses": {
+          "202": {
+            "description": "Accepted the request to delete the addon."
+          },
+          "204": {
+            "description": "The addon is already deleted."
+          },
+          "200": {
+            "description": "Successfully deleted the addon."
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-long-running-operation": true,
```

Comment:
Doesn't the API return 'Location' header to track async DELETE?

Reply:
It returns azure async repsonse header which is tracked.

Reply:
Then please add `"x-ms-long-running-operation-options" : { "final-state-via" : "location" }` By default it assumes using Azure-AsyncOperation header.
https://github.com/Azure/autorest/blob/master/docs/extensions/readme.md#x-ms-long-running-operation

Reply:
Changed as requested.